### DISCUSSION TITLE
will it support oss from aliyun and obs  from huawei yun

### DISCUSSION THREAD

**Discussion by asmlgkj** [nextflow-io/nextflow#3043] [Q&A] [answered]
Dear developer,
Thanks a lot for such great work, will it support other cloud company service other than aws, such as alibaba cloud and huawei cloud

**Comment by bentsherman** [comment, marked as answer]
Hi @asmlgkj , see this related Q&A: #2795

Basically, we aren't planning to add support for these cloud platforms ourselves anytime soon, but anyone is welcome to do it. You could adapt one of the existing executors e.g. `awsbatch` or `azurebatch` to work with a different cloud provider as long as it has a Java library or REST API.

Additionally, you can use these clouds now through Kubernetes -- simply set up a K8s cluster in a cloud environment and use Nextflow's `k8s` executor. It's not as convenient as a cloud batch executor, but it's generic enough that it can work with any platform that supports K8s.

**Reply by asmlgkj** [reply]
Thanks a lot
